scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala

```diff
@@ -63,7 +66,13 @@ final class DynamicBigQueryOps[T](private val self: SCollection[T]) extends AnyV
       if (writeDisposition != null) {
         transform = transform.withWriteDisposition(writeDisposition)
       }
-      self.applyInternal(transform)
+      val wr = self.applyInternal(transform)
+      if (self.internal.isBounded() == IsBounded.UNBOUNDED) {
+        insertErrorTransform(self.context.wrap(wr.getFailedInsertsWithErr()))
```

Comment:
I haven't tested this but from what I read, I think we have to call [`withExtendedErrorInfo`](https://beam.apache.org/releases/javadoc/2.11.0/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.Write.html#withExtendedErrorInfo--) first to make BQ write extended info and then we can access [`getFailedInsertsWithErr`](https://beam.apache.org/releases/javadoc/2.11.0/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.html#getFailedInsertsWithErr--)